lib/AL_USDMaya/AL/usdmaya/Global.cpp

```diff
@@ -153,8 +208,8 @@ void Global::onPluginLoad()
   m_fileNew = MSceneMessage::addCallback(MSceneMessage::kAfterNew, onFileNew);
   m_preSave = MSceneMessage::addCallback(MSceneMessage::kBeforeSave, preFileSave);
   m_postSave = MSceneMessage::addCallback(MSceneMessage::kAfterSave, postFileSave);
-  m_preOpen = MSceneMessage::addCallback(MSceneMessage::kBeforeOpen, preFileOpen);
-  m_postOpen = MSceneMessage::addCallback(MSceneMessage::kAfterOpen, postFileOpen);
+  m_preRead = MSceneMessage::addCallback(MSceneMessage::kBeforeFileRead, preFileRead);
```

Comment:
No, this should work fine - we control whether assets are brought in as usd/hydra or MayaReference types with variants in our own pipe, and it hasn't caused problems.  In theory, if your referenced scene in turn had it's own ProxyShape, you'd want this to trigger anyway (though I wouldn't really recommend doing that, it's not really heavily tested).

Reply:
Oh, and regarding layer.h - it's there just to provide a stub implementation of the layer nodes.  They no longer do anything, but the node is still provided by the plugin so that if you open a scene that was saved using old-style-layer serialization, it won't error - all the same attributes, etc are there.

Of course, any serialized layer changes you'd made in the old format will no longer be recognized / will be lost.  At some point in the future, we could potentially write some upgrade / transition code, which reads the old layer nodes.  Actually, probably not even be hard to do... would just had to add a loop inside of the postRead callback that loads all the layers...

Reply:
Sounds good about the references not being a problem still!

Sure, no harm in leaving the node code there I guess, we will remove it at some point.

Cheers!

Reply:
Oh, and regarding the mutex locks - as far as I know, we're not loading layers in a multithreaded manner.  The locks are more from paranoia... I don't know they COULDN'T be loaded in a multithreaded manner (I haven't really looked into the way maya's new multi-threaded node evaluation works, for instance), and it's essentially a globally shared resource, so I figured better be safe.